test/contracts/LLL_ENS.cpp

```diff
@@ -349,6 +349,7 @@ class LLLENSTestFramework: public LLLExecutionFramework
 			BOOST_REQUIRE(errors.empty());
 		}
 		sendMessage(*s_compiledEns, true);
+		BOOST_REQUIRE(m_transactionStatus);
 		BOOST_REQUIRE(!m_output.empty());
```

Comment:
I think it would be time to introduce a helper for `ensureSuccessfulTransaction()` and `ensureSuccessfulTransactionWithData()` (please chose better names). Maybe we can change `sendMessage` to have flags (enums?) or multiple versions of `sendMessage` for that.

At the bare minimum I think instead of `BOOST_REQUIRE(m_transactionStatus)` we should have something like `BOOST_REQUIRE(m_transactionSuccessfull)`.

Reply:
In general, I agree, but I'm not sure ``sendMessage`` is the right place for that. ``sendMessage`` is rather low-level and only used at very few places outside ``ExecutionFramework`` and ``SolidityExecutionFramework``. It also doesn't return ``m_output``, but only sets it implicitly, so I think it's fine if it does the same for ``m_transactionStatus``.

I think the best model is ``compileAndRun`` and ``compileAndRunWithoutCheck`` in ``ExecutionFramework``. I already added a ``BOOST_REQUIRE(m_transactionStatus);`` to the former. I'd suggest that similarly all the ``call`` functions in ``ExecutionFramework`` should have a checked and unchecked version. Or alternatively they should always check ``m_transactionStatus`` to be true and there should be another version to be used when expecting failure that asserts ``m_transactionStatus`` to be false and doesn't return ``bytes``.

In any case I think it makes more sense to keep this PR simple and to only introduce these distinctions after #4245, since otherwise it will result in duplicated work for semantics tests that can be extracted.

Reply:
Thinking about it, it may not be too much work to let ``call`` verify ``m_transactionStatus`` to be true and just replace everything that then fails with a new ``callAndExpectFailure`` (or similar) function, so I can already do that in this PR, if you think that makes sense?